nntrainer/layers/conv2d_layer.cpp

```diff
@@ -128,9 +128,19 @@ void Conv2DLayer::forwarding(sharedConstTensors in) {
     Tensor out = hidden_.getBatchSlice(b, 1);
     Tensor inSub = input_.getBatchSlice(b, 1);
 
-    status =
-      conv2d_gemm(filter_kernel.getData(), filter_kernel.getDim(), inSub,
-                  out_dim, stride, padding, out.getData(), out.length(), true);
+    std::vector<float> in_col;
+
+    in_col.resize(filter_kernel.getDim().getFeatureLen() * out_dim.width() *
+                  out_dim.height());
+
+    Tensor in_padded = zero_pad(0, inSub, padding.data());
```

Comment:
Please update zero_pad function signature to take output tensor as input. 
Then, if padding is zero, lets no copy but directly use the input. 
Currently, memory copy happens when padding is 0 inducing significant overhead.

Reply:
done.